File: apps/rayo/scripts/read-excel-rco005.mjs
import fs from 'fs';
import * as xlsx from 'xlsx';

const filePath = '/Users/ryanrichard/projecont/Rayo/temp/braga/dados validados/RCO005_PlanoConta (1).xlsx';
const workbook = xlsx.readFile(filePath);
const sheetName = workbook.SheetNames[0];
const worksheet = workbook.Sheets[sheetName];
const data = xlsx.utils.sheet_to_json(worksheet, { header: 1 });

// Procurando a coluna com a conta formatada (ex: 6.1.1.01.002) e a conta Dealer de 12 dígitos
console.log('--- AMOSTRA DAS PRIMEIRAS 10 LINHAS ---');
for (let i = 0; i < Math.min(10, data.length); i++) {
  console.log(`Row ${i}:`, data[i]);
}

// Tentativa heurística de achar a coluna RCO005 (formato com pontos) e a de 12 dígitos
const mappings = {};
let foundCount = 0;

data.forEach(row => {
  if (!Array.isArray(row)) return;
  
  // Extrai strings da linha
  const strings = row.map(cell => String(cell || '').trim());
  
  // Procura uma conta formato Fortes (com pontos)
  const fortesAcc = strings.find(s => /^\d(\.\d+)+$/.test(s));
  
  // Procura uma conta formato Dealer (12 dígitos numéricos)
  const dealerAcc = strings.find(s => /^\d{12}$/.test(s));
  
  if (fortesAcc && dealerAcc) {
    mappings[fortesAcc] = dealerAcc;
    foundCount++;
  }
});

console.log('\n--- MAPA ENCONTRADO (Heurística: Formato Fortes -> 12 Dígitos) ---');
console.log(`Encontrados ${foundCount} mapeamentos possíveis.`);
if (foundCount > 0) {
  const sample = Object.entries(mappings).slice(0, 5);
  console.log('Amostra de mapeamento:', sample);
} else {
  console.log('Não foi possível achar o padrão automático na mesma linha.');
  console.log('Por favor verifique as colunas manualmente na amostra de dados.');
}

File: apps/rayo/scripts/read-excel-rco005.py
import pandas as pd
import json

file_path = '/Users/ryanrichard/projecont/Rayo/temp/braga/dados validados/RCO005_PlanoConta (1).xlsx'

try:
    df = pd.read_excel(file_path)
    
    mappings = {}
    found_count = 0
    
    # Heurística: procurar em todas as linhas e colunas
    for index, row in df.iterrows():
        fortes_acc = None
        dealer_acc = None
        
        for col in df.columns:
            val = str(row[col]).strip()
            
            # Formato Fortes: X.X.X.XX.XXX
            if len(val.split('.')) >= 4 and all(part.isdigit() for part in val.split('.')):
                fortes_acc = val
            
            # Formato Dealer: 12 dígitos
            if len(val) == 12 and val.isdigit():
                dealer_acc = val
        
        if fortes_acc and dealer_acc:
            mappings[fortes_acc] = dealer_acc
            found_count += 1
            
    print("--- MAPA ENCONTRADO ---")
    print(f"Total: {found_count}")
    print(json.dumps(mappings, indent=2))
    
except Exception as e:
    print(f"Erro ao ler excel: {e}")
